### PR TITLE
feat: add actions to component specification

### PR DESCRIPTION
**Pull request by AlexKlim**
## Description

Add "actions" to components spec

## Issues

Please paste the links of related issues about this PR.

## Videos

Video to prove your work

## Test

What kind of tests were performed and what were the results?

# Checklist

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation to reflect the change
- [ ] I have updated the README.md to reflect the change
- [ ] I have added unit tests to prove that the update is effective and the feature works

### CODE REVIEW

File: node/lib/protos/v2/component_specification.proto.ts [hunk 1]
```diff
@@ -1,26 +1,34 @@
 /* eslint-disable */
 import * as _m0 from "protobufjs/minimal";
 
-export class ComponentSpecification {
+export interface ComponentSpecification {
```
Comment (ak15) [inline node/lib/protos/v2/component_specification.proto.ts:4]: I am thinking why its interface ? 🤔

File: node/lib/protos/v2/component_specification.proto.ts [hunk 2]
```diff
@@ -1,26 +1,34 @@
...
+  referenceId: string;
+}
+
 function createBaseComponentSpecification(): ComponentSpecification {
-  return { type: "", title: "", description: "", figmaNodeId: "" };
+  return { type: "", title: "", description: "", figmaNodeId: "", actions: [] };
 }
 
-export const ComponentSpecificationData = {
+export const ComponentSpecification = {
```
[the remark this answers is not in the document]
Comment (AlexKlim) [reply node/lib/protos/v2/component_specification.proto.ts:31]: I run from Readme: 
```bash
cd node
yarn protobuf:generate
```

File: node/lib/protos/v2/component_specification.proto.ts (hunk 1 above)
Comment (AlexKlim) [reply node/lib/protos/v2/component_specification.proto.ts:4]: automatically change. 
by the script in the Readme

**Comment by zrg-team** [comment]
@eebasadre20 you met this issue previously right ?

File: node/lib/protos/v2/component_specification.proto.ts (hunk 1 above)
Comment (eebasadre20) [reply node/lib/protos/v2/component_specification.proto.ts:4]: @AlexKlim same comment as Atul.

File: node/lib/protos/v2/component_specification.proto.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (eebasadre20) [reply node/lib/protos/v2/component_specification.proto.ts:31]: @AlexKlim can you try to check the 'Known Issue' in root README.md

[the remark this answers is not in the document]
Comment (AlexKlim) [reply node/lib/protos/v2/component_specification.proto.ts:31]: @eebasadre20 yes I did 
let me generate one more time and see what happens

**Comment by AlexKlim** [comment]
@eebasadre20 @ak15 @akuita @zrg-team 
I did: 
```bash
cd node 
rm -rf node_modules
yarn install
yarn protobuf:generate
```

looks like it generates as needed. Please take a look

File: protos/v2/component_specification.proto [hunk 3]
```diff
@@ -7,11 +7,19 @@ message ComponentSpecification {
...
     float top = 1;
     float left = 2;
     float right = 3;
     float bottom = 4;
   }
+
+  message Action {
+    string type = 1;
+    string description = 2;
+    string reference = 3;
```
Comment (akuita) [inline protos/v2/component_specification.proto:22]: should it be `reference_type`?

**Review by akuita — APPROVED** [review]

**Review by ak15 — APPROVED** [review]
👍

File: protos/v2/component_specification.proto (hunk 3 above)
Comment (AlexKlim) [reply protos/v2/component_specification.proto:22]: @zrg-team do you prefer `reference_type` instead of `reference`? 
both variants work for me

Comment (zrg-team) [reply protos/v2/component_specification.proto:22]: reference type maybe more clear. It will be api or bl, use case. screen

**Review by zrg-team — APPROVED** [review]

**Pull request merged**